modAionImpl/src/org/aion/zero/impl/pendingState/AionPendingStateImpl.java

```diff
@@ -429,12 +398,59 @@ else if (bestRepoNonce(txFrom).compareTo(txNonce) < 1) {
         return txResponses;
     }
 
-    private boolean inPool(BigInteger txNonce, AionAddress from) {
-        return (this.txPool.bestPoolNonce(from).compareTo(txNonce) > -1);
+    private void addPendingTxToBackupDatabase(AionTransaction tx) {
+        if (poolBackUpEnable) {
+            backupPendingPoolAdd.put(tx.getTransactionHash(), tx.getEncoded());
+        }
+    }
+
+    private void removeBackupDBPendingTx(byte[] txHash) {
+        if (poolBackUpEnable) {
+            backupPendingPoolRemove.add(txHash);
+        }
+    }
+
+    private int calculateTxFetchNumberLimit() {
+        int cachedAccount = pendingTxCache.getCacheTxAccount().size();
+        return cachedAccount == 0 ? 1 : Math.max((txPool.maxPoolSize / 4) / cachedAccount, 1);
+    }
+
+    private TxResponse addTransactionToCache(AionTransaction tx, boolean transactionPoolIsFull) {
+
+        if (pendingTxCache.addCacheTx(tx) == null) {
+            LOGGER_TX.debug(
```

Comment:
Instead of logging every option, could we log only the result from the method calling this? The `TxResponse` already has a flag for failure, so it would be more reliable to use that instead of relying on the text being correctly maintained. Also, if additional information is required, the string could be added to the `TxResponse`.